ArjunaJTA/jta/classes/com/arjuna/ats/jta/logging/jtaLogger.java

```diff
@@ -33,8 +33,47 @@
 
 import org.jboss.logging.Logger;
 
-public class jtaLogger
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+public class jtaLogger implements InvocationHandler
 {
     public static final Logger logger = Logger.getLogger("com.arjuna.ats.jta");
-    public static final jtaI18NLogger i18NLogger = Logger.getMessageLogger(jtaI18NLogger.class, "com.arjuna.ats.jta");
+    public static final jtaI18NLogger i18NLogger = (jtaI18NLogger) Proxy.newProxyInstance(
+            jtaI18NLogger.class.getClassLoader(),
+            new Class[] { jtaI18NLogger.class },
+            new jtaLogger(Logger.getMessageLogger(jtaI18NLogger.class, "com.arjuna.ats.jta")));
+
+    private jtaI18NLogger jtaI18NLoggerImpl;
+    private static boolean recoveryProblems;
+
+    private jtaLogger() {
+    }
+
+    private jtaLogger(jtaI18NLogger logger) {
+        jtaI18NLoggerImpl = logger;
+    }
+
+    public static boolean isRecoveryProblems() {
+        return recoveryProblems;
+    }
+
+    public static void setRecoveryProblems(boolean recoveryProblems) {
+        jtaLogger.recoveryProblems = recoveryProblems;
```

Comment:
Just noticed that no-one is resetting this (perhaps at start of periodic recovery or at the phaseOne in XARM)